### ISSUE TITLE
CTCLoss cuda error

### ISSUE THREAD

**Issue by swethmandava** [pytorch/pytorch#16093]
## To Reproduce

from torch.nn import CTCLoss
criterion = CTCLoss(reduction="none")

nn_loss = criterion(softmax_out, targets, output_seq_len, target_sizes)
nn_loss = nn_loss.sum()
optimizer.zero_grad()
nn_loss.backward(retain_graph=True)
nn_norm = get_norm(model.parameters())

nn_loss2 = criterion(softmax_out.cpu(), targets, output_seq_len, target_sizes)
nn_loss2 = nn_loss2.sum()
optimizer.zero_grad()
nn_loss2.backward()
nn_norm2 = get_norm(model.parameters())

When I try this, nn_loss2 = nn_loss but nn_norm2 != nn_norm. I compared it to gradients obtained using warpctc as well. There seems to be a bug in the cuda implementation of CTC Loss

## Environment

Collecting environment information...
PyTorch version: 1.0.0a0
Is debug build: No
CUDA used to build PyTorch: 10.0.130

OS: Ubuntu 16.04.5 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609
CMake version: version 3.5.1

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration: GPU 0: Quadro GV100
Nvidia driver version: 410.48
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.4.1

Versions of relevant libraries:
[pip] Could not collect
[conda] magma-cuda100             2.1.0                         5    local
[conda] mkl                       2019.1                      144  
[conda] mkl-include               2019.1                      144  
[conda] torch                     1.0.0a0                   <pip>
[conda] torchaudio                0.2                       <pip>
[conda] torchtext                 0.4.0                     <pip>
[conda] torchvision               0.2.1                     <pip>
[conda] warpctc-pytorch           0.1                       <pip>

**Comment by soumith** [comment]
are you on latest master?
CPU norm on 1.0.0 had a bug: https://github.com/pytorch/pytorch/issues/15602
It was fixed in this PR: https://github.com/pytorch/pytorch/pull/15885

This patch will go into 1.0.1 this week.
